core/trino-main/src/main/java/io/trino/cost/FilterStatsCalculator.java

```diff
@@ -352,14 +352,15 @@ protected PlanNodeStatsEstimate visitComparisonExpression(ComparisonExpression n
             Expression left = node.getLeft();
             Expression right = node.getRight();
 
-            checkArgument(!(left instanceof Literal && right instanceof Literal), "Literal-to-literal not supported here, should be eliminated earlier");
+            checkArgument(!(isEffectivelyLiteral(left) && isEffectivelyLiteral(right)), "Literal-to-literal not supported here, should be eliminated earlier");
```

Comment:
Where is this case handled? Is it SimplifyExpressions?

Reply:
in `FilterStatsCalculator`, https://github.com/trinodb/trino/blob/8e40a44b977643352e69d4f4356aa65f4bad0bf6/core/trino-main/src/main/java/io/trino/cost/FilterStatsCalculator.java#L102-L120